pulser-core/pulser/waveforms.py

```diff
@@ -905,6 +903,14 @@ def _err_message(argument_name: str) -> str:
                 raise ValueError(
                     "`times` must be an array of non-repeating values."
                 )
+        else:
+            times_ = times
+        if times_.size != values_.size:
```

Comment:
Maybe you can do this:
```suggestion
        try:
            size_mismatch = (times_.size != values_.size)
        except AttributeError:
            size_mismatch = False
        if size_mismatch:
```

Reply:
I think it's going to be a nightmare for mypy, the fact that no one had an issue over the last 4 months makes me think that it's overkilled to make a too complex testing

Reply:
Ok, I'm fine with what you have right now